string got_cmd
bool executing
bool done